vllm/v1/sample/rejection_sampler.py

```diff
@@ -81,30 +135,51 @@ def flashinfer_sample(
-                                            batch_first=True,
-                                            padding_value=INVALID_TOKEN_ID)
-
-            vocab_size = target_probs.size(-1)
-            # NOTE: CPU <-> GPU synchronization happens here.
-            draft_token_ids_tensor = draft_token_ids_tensor.to(
-                target_probs.device)
-            draft_probs = _create_greedy_token_probs(draft_token_ids_tensor,
-                                                     vocab_size,
-                                                     target_probs.device)
-            target_probs = _create_greedy_token_probs(target_token_ids,
-                                                      vocab_size,
-                                                      target_probs.device)
-            uniform_samples = torch.zeros(draft_token_ids_tensor.size(0),
-                                          draft_token_ids_tensor.size(1) + 1,
-                                          device=target_probs.device)
-        else:
-            raise NotImplementedError(
-                "Currently, only greedy sampling is supported by "
-                "rejection sampler.")
+        # NOTE: CPU <-> GPU synchronization happens here.
+        draft_token_ids_tensor = draft_token_ids_tensor.to(target_probs.device)
+
+        if self.forward_method == self.flashinfer_sample:
+            # Create one-hot tensor for draft token ids.
+            # This is used for ngram where we don't have draft_probs.
+            if draft_probs is None:
+                vocab_size = target_probs.size(-1)
+                draft_probs = _create_greedy_token_probs(
+                    draft_token_ids_tensor, vocab_size, target_probs.device)
+            if sampling_metadata.all_greedy:
+                target_token_ids_tensor = target_probs.argmax(dim=-1)
+                target_probs = _create_greedy_token_probs(
+                    target_token_ids_tensor, vocab_size, target_probs.device)
+            else:
+                sample_lens = [len(x) + 1 for x in draft_token_ids]
+                target_probs = _convert_2d_probs(target_probs, sample_lens)
+
+        elif self.forward_method == self.forward_native:
+            # Create one-hot tensor for draft token ids.
+            # This is used for ngram where we don't have draft_probs.
+            if draft_probs is None and not sampling_metadata.all_greedy:
+                vocab_size = target_probs.size(-1)
+                draft_probs = _create_greedy_token_probs(
+                    draft_token_ids_tensor, vocab_size, target_probs.device)
+            sample_lens = [len(x) + 1 for x in draft_token_ids]
+            target_probs = _convert_2d_probs(target_probs, sample_lens)
+
+        return self.forward_method(draft_token_ids_tensor, draft_probs,
+                                   bonus_token_ids_tensor, target_probs,
+                                   sampling_metadata)
+
+    def flashinfer_sample(
```

Comment:
Given that we will have our own custom kernel, I think it's ok to remove the flashinfer path in this PR.